### PR TITLE
SDK-1752: Java SDK message passing docs

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
## What does this PR do?
 * Copied message-passing section from https://github.com/temporalio/documentation/pull/2976 (as of when this branch was created) and altered for Java
 * Replaced most uses of `async` in docs with blocking, kept headers for cross language consistency but open to change
 ## Notes to reviewers

Internal Jira link: https://temporalio.atlassian.net/browse/SDK-1752

### CODE REVIEW

**Comment by fairlydurable** [comment]
This appears to be https://temporalio.atlassian.net/browse/SDK-1752
Which will eventually become https://temporalio.atlassian.net/browse/EDU-670

Please let me know if I've mis-identified the PR draft.

**Review by dandavison — APPROVED** [review]
Awesome, thanks very much @Quinn-With-Two-Ns. I pushed two commits to brind up to date with the Python docs.

Over to you @fairlydurable!

File: docs/develop/python/message-passing.mdx [hunk 1]
```diff
@@ -396,31 +396,26 @@ class GreetingWorkflow:
...
-Although an `async def` Signal handler can initiate similar network tasks, using an Update handler allows the client to receive a result or error once the Activity completes.
+Although an `async def` Signal handler can also execute an Activity, using an Update handler allows the client to receive a result or error once the Activity completes.
 This lets your client track the progress of asynchronous work performed by the Update's Activities, Child Workflows, etc.
 
 ### Add wait conditions to block
 
 Sometimes, `async def` Signal or Update handlers need to meet certain conditions before they should continue.
-You can use a wait condition ([`workflow.wait_condition`](https://python.temporal.io/temporalio.workflow.html#wait_condition)) to set a function that prevents the code from proceeding until the condition returns `True`.
+You can use [`workflow.wait_condition`](https://python.temporal.io/temporalio.workflow.html#wait_condition) to prevent the code from proceeding until a condition is true.
+You specify the condition by passing a function that returns `True` or `False`.
```
Comment (angelazhou32) [inline docs/develop/python/message-passing.mdx:406]: Do you want to mention the timeout?

Comment (dandavison) [reply docs/develop/python/message-passing.mdx]: Good call! Done.

**Review by fairlydurable — APPROVED** [review]
@angelazhou32 Performed team review.
@dandavison responded to Angela's review.

I performed a light sanity check after this process.

Stamping.

File: docs/develop/java/message-passing.mdx [hunk 2]
```diff
@@ -27,201 +27,261 @@ tags:
...
+            }
+        }
+    }
 
-**How to handle Signals in a Workflow using the Java SDK.**
+}
+```
 
-Workflows listen for Signals by the Signal's name.
+- A Query handler must not modify workflow state.
```
Comment (fairlydurable) [inline docs/develop/java/message-passing.mdx:94]:
```suggestion
- A Query handler must not modify Workflow state.
```

File: docs/develop/java/message-passing.mdx [hunk 3]
```diff
@@ -27,201 +27,261 @@ tags:
...
 
-This Workflow completes when the Signal updates the greeting to `Bye`.
+- The Update annotation accepts arguments (`name`, and `unfinished_policy`) as described in the API reference docs for  [`@UpdateMethod`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/workflow/UpdateMethod.html).
 
-### Send a Signal from a Temporal Client {#send-signal-from-client}
+- About validators:
+  - Use validators to reject an Update before it is written to History.
+    Validators are always optional.
+    If you don't need to reject Updates, you can skip them.
+  - Define an update validator with the `@UpdateValidatorMethod` annotation.
```
Comment (fairlydurable) [inline docs/develop/java/message-passing.mdx:187]:
```suggestion
  - Define an Update validator with the `@UpdateValidatorMethod` annotation.
```

File: docs/develop/java/message-passing.mdx [hunk 4]
```diff
@@ -256,170 +308,333 @@ public interface GreetingWorkflow {
...
 ```
 
-Note that the Signal handler `setCustomer` is executed before the `@WorkflowMethod` `greet` is called.
+### Send an Update {#send-update-from-client}
 
-## Queries {#queries}
+An Update is a synchronous, blocking call that can change Workflow state, control its flow, and return a result.
 
-A [Query](/encyclopedia/workflow-message-passing#sending-queries) is a synchronous operation that is used to get the state of a Workflow Execution.
+A client sending an Update must wait until the Server delivers the Update to a Worker.
```
Comment (fairlydurable) [inline docs/develop/java/message-passing.mdx:322]:
```suggestion
A Client sending an Update must wait until the Server delivers the Update to a Worker.
```

File: docs/develop/java/message-passing.mdx [hunk 5]
```diff
@@ -256,170 +308,333 @@ public interface GreetingWorkflow {
...
 
-**How to send a Query to a Workflow Execution from a Temporal Client using the Java SDK.**
+When sending a Signal, Update, or Query to a Workflow, your Client might encounter the following errors:
 
-Queries are sent from a Temporal Client.
+- **The client can't contact the server**:
+  You'll receive a [`WorkflowServiceException`](https://javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/client/WorkflowServiceException.html) on which the `cause` is a [`StatusRuntimeException`](https://grpc.github.io/grpc-java/javadoc/io/grpc/StatusRuntimeException.html) and `status` of `UNAVAILABLE` (after some retries).
 
-To send a Query to a Workflow Execution from an external process, call the Query method (defined in the Workflow) from a `WorkflowStub` within the Client code.
+- **The workflow does not exist**:
```
Comment (fairlydurable) [inline docs/develop/java/message-passing.mdx:553]:
```suggestion
- **The Workflow does not exist**:
```

**Review by fairlydurable — CHANGES_REQUESTED** [review]
Light passes on all of these due to time limits

* Spellcheck, grammar check
* Caps check
* Overall inspection

**Review by fairlydurable — APPROVED** [review]

**Pull request merged**